torch/utils/_pytree.py

```diff
@@ -422,18 +450,31 @@ def _namedtuple_unflatten(values: Iterable[Any], context: Context) -> NamedTuple
 
 
 def _namedtuple_serialize(context: Context) -> DumpableContext:
-    json_namedtuple = {
-        "class_name": context.__name__,
-        "fields": context._fields,
-    }
-    return json_namedtuple
+    if context not in SUPPORTED_SERIALIZED_TYPES:
+        raise NotImplementedError(
+            f"Serializing namedtuple class {context} in pytree is not "
+            "registered. Please register using `_register_namedtuple`",
```

Comment:
nit: "Can't serialize TreeSpec of namedtuple class {context} because we didn't register a serializated_type_name. Please register using `_register_namedtuple`